packetbeat/tests/system/packetbeat.py

```diff
@@ -11,6 +11,11 @@
 TRANS_REQUIRED_FIELDS = ["@timestamp", "type", "status",
                          "agent.type", "agent.hostname", "agent.version"]
 
+TRANS_ECS_REQUIRED_FIELDS = TRANS_REQUIRED_FIELDS + ["source.ip",
+                                                     "destination.ip", "event.dataset", "event.start",
+                                                     "network.type", "network.transport", "network.protocol",
+                                                     "network.community_id", "client.ip", "server.ip"]
```

Comment:
`event.end` and `event.duration` as well?

Reply:
There are some tests where the end of the transaction is missed so we don't report `event.end` and then without end we can't compute `event.duration`. Another case where `event.end` is not reported is for the non-transacted protocols implementations like ICMP and DHCPv4.

There may be tests were we miss the start (request) and only observe the response (`event.end`). So I may need to make a few more changes to account for these edge cases. But I'll handle those as exceptions rather than minimizing the amount of fields we check for in the common case.

Reply:
I pushed a change to add `event.end` and `event.duration` because they are present in all of the pgsql test cases. I'll adjust it as needed when I get done (BTW just one more protocol to do - memcached).